components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java

```
@@ -1015,8 +1015,12 @@ public void updateSubscriptionStatus(String subId, APIMgtConstants.SubscriptionS
             APIManagementException {
         try {
             getApiSubscriptionDAO().updateSubscriptionStatus(subId, subStatus);
+            Subscription subscription = getSubscriptionByUUID(subId);
+            getApiGateway().updateAPISubscriptionStatus(subscription);
         } catch (APIMgtDAOException e) {
             throw new APIManagementException(e);
+        } catch (GatewayException e) {
+            log.error("Error occurred while updating subscription status in gateway", e);
```

this is not need to throw the error
better to have a comment